pkg/cspc/algorithm/build_csp.go

```diff
@@ -0,0 +1,103 @@
+/*
+Copyright 2020 The OpenEBS Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package algorithm
+
+import (
+	"github.com/openebs/maya/pkg/version"
+	"github.com/pkg/errors"
+	cstor "github.com/openebs/api/pkg/apis/cstor/v1"
+	"github.com/openebs/api/pkg/apis/types"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/rand"
+)
+
+const (
+	// StoragePoolKindCSPC holds the value of CStorPoolCluster
+	StoragePoolKindCSPC = "CStorPoolCluster"
+	ApiVersion= "cstor.openebs.io/v1"
+)
+
+// GetCSPSpec returns a CSPI spec that should be created and claims all the
+// block device present in the CSPI spec
+func (ac *Config) GetCSPSpec() (*cstor.CStorPoolInstance, error) {
+	poolSpec, nodeName, err := ac.SelectNode()
+	if err != nil || nodeName == "" {
+		return nil, errors.Wrap(err, "failed to select a node")
```

Comment:
This is dangerous if err is nil and nodeName is "" then an error will be swallowed. Symantec wise doesn't looks good.